src/DankCord/core.py

```diff
@@ -86,74 +89,33 @@ def _run_command(self, name: str, retry_attempts= 3, timeout:int = 10):
-            if errors:
-                # TODO horrible way, sometimes the error does NOT have data, so this needs fixing
-                if response.data["errors"]["data"]["_errors"][0]["code"] == "COMPONENT_VALIDATION_FAILED":
-                    raise InvalidComponent("Invalid component.")
-                if retry_after > 0:
-                    self.logger.log(level="Warning", msg=f"Ratelimited while running {name}, retrying after {retry_after}")
-                    time.sleep(retry_after)
-
-                return None
-            if code == 10003:
-                raise UnknownChannel("Bot doesn't have access to this channel.")
-
-        lim = time.time() + timeout
-        if message_id == "":
-            while time.time() < lim:
-                if nonce in self.ws_cache:
-                    for i in event_type:
-                        if i in self.ws_cache[nonce]:
-                            _message = self.ws_cache[nonce][i]
-                            return Message(_message)
-
-        if not _message and message_id == "":
-            raise NonceTimeout("Did not receive nonce in time.")
-
-        lim = time.time() + timeout
-        while time.time() < lim:
             try:
-                for key, value in self.ws_cache.items():
-                    try:
-                        event = key.split(" ")[1]
-                        if value["id"].strip() == message_id.strip() and event in event_type:
-                            _message = value
-                            return Message(_message)
-                        if (
-                            value["message_reference"]["message_id"].strip() == message_id.strip
-                            and value["type"] == 19
-                            and event in event_type
-                        ):
-                            _message = value
-                            return Message(_message)
-                    except Exception as e:
-                        pass
-            except:
-                pass
-
-        if not _message:
-            raise NonceTimeout("Did not receive nonce in time.")
+                interaction: Optional[Union[Message, bool, None]] = self.wait_for("MESSAGE_CREATE", nonce)
+                return interaction
+            except Exception as e:
+                print(f"Error in core.py _run_command: {e}")
+                continue
+            
+        return None
 
+    def wait_for(
+        self,
+        event: Literal["MESSAGE_CREATE", "MESSAGE_UPDATE", "INTERACTION_CREATE", "INTERACTION_SUCCESS"],
+        nonce_or_id: str,
+        timeout: int = 10
+    ) -> Optional[Union[Message, bool, None]]:
+        limit = time.time() + timeout
+        while time.time() < limit:
+            cache = self.gateway.cache
+            if event in ("INTERACTION_CREATE", "INTERACTION_SUCCESS"):
+                if nonce_or_id in cache.interaction_create or nonce_or_id in cache.interaction_success:
```

Comment:
It should be more specific to the exact `event` param. An interaction may have created but not succeeded which would result in a false positive